src/Monticello/MethodAddition.class.st

```diff
@@ -44,11 +44,14 @@ MethodAddition >> createCompiledMethod [
 	"CyrilFerlicot: Why do we need to explicitly write to the source? Why do we call the compiler?
 	
 	Can't we just call #compile:classfied: here?"
-	compiledMethod := myClass compiler permitUndeclared: true; compile: text asString.
+
+	compiledMethod := myClass compiler
+		                  permitUndeclared: true;
+		                  compile: text asString.
 	selector := compiledMethod selector.
 	self writeSourceToLog.
 	priorMethodOrNil := myClass compiledMethodAt: selector ifAbsent: [ nil ].
-	priorCategoryOrNil := myClass organization protocolNameOfElement: selector
+	priorCategoryOrNil := (myClass organization protocolOfSelector: selector) ifNotNil: [ :protocol | protocol name ]
```

Comment:
Well, ideally I want to iterate further to use real protocol in most part of the system and not manipulate the protocol name. I'm just doing it step by step and for now I want to kill the users of the hacked #protocolNameOfElement:  

I wanted to check once I would use real protocols in most places if we still had a lot of places that needed to deal with protocol names instead of protocols to see if it makes sense to introduce a #protocolNameOfSelector: that does not have the hack that #protocolNameOfElement: has